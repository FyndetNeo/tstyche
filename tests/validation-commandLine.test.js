import { afterEach, describe, expect, test } from "@jest/globals";
import { clearFixture, writeFixture } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixture = "validation-commandLine";

afterEach(async () => {
  await clearFixture(fixture);
});

describe("'tstyche' command", () => {
  test.each([
    {
      args: ["--cache"],
      testCase: "handles unknown command line option in a long form",
    },
    {
      args: ["-c"],
      testCase: "handles unknown command line option in a short form",
    },
  ])("$testCase", async ({ args }) => {
    await writeFixture(fixture);

    const { exitCode, stderr, stdout } = await spawnTyche(fixture, args);

    expect(stdout).toBe("");
    expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

    expect(exitCode).toBe(1);
  });
});
