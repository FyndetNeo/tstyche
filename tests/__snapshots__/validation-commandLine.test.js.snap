// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'tstyche' command handles unknown command line option in a long form: stderr 1`] = `
"Error: Unknown option '--cache'.

"
`;

exports[`'tstyche' command handles unknown command line option in a short form: stderr 1`] = `
"Error: Unknown option '-c'.

"
`;
